sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -339,7 +339,9 @@ case class DataSource(
         val baseRelation =
           dataSource.createRelation(sparkSession.sqlContext, caseInsensitiveOptions)
         if (baseRelation.schema != schema) {
-          throw new AnalysisException(s"$className does not allow user-specified schemas.")
+          throw new AnalysisException(s"$className does not allow user-specified schemas, " +
+              s"user-specified schemas are not same as source schema: \n" +
```

Comment:
I am not against this change but you gotta make this error message as below at least.

```
s"$className does not allow user-specified schemas because user-specified" +
s"schema [${schema.catalogString}] was different from its source schema" +
s"[${baseRelation.schema.catalogString}]"
```

Reply:
OK, Thanks for @HyukjinKwon 's replies.